### PR TITLE
Dlt jeklis

### PR DESCRIPTION
**Pull request by cann0010**
I propose pulling this branch into main. The primary changes are:
(1) This uses Nathan's "Docker local theme" version, so it can be run in a local Docker instance. The theme is loaded as a submodule.
(2) A simplified navbar menu reduces it to the items that I think I'd like in the new site.
(3) Code for dropdown items has been added in _includes/navbar-items.html
(4) Data for genus pages has been added, in files genus_resources.yml and species_resources.yml, for each GENUS in _data/taxa/
(5) A generator script has been added to _plugins. This iterates over the GENUS directories in _data/taxa/ to generate the index.html files at taxa/GENUS

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "_themes/legumeinfo-jekyll-theme"]
+	path = _themes/legumeinfo-jekyll-theme
+	url = https://github.com/legumeinfo/legumeinfo-jekyll-theme.git
```
Comment (alancleary) [inline .gitmodules:3]: Before this goes into production it should be updated to track a commit associated with a specific tag in the theme repo:
```bash
cd _themes/legumeinfo-jekyll-theme
git checkout v1.0.0
cd ..
git add legumeinfo-jekyll-theme
git commit -m "Moved theme submodule to v1.0.0"
git push
```

File: _plugins/jekyll-generate-taxa.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env ruby
```
Comment (alancleary) [inline _plugins/jekyll-generate-taxa.rb:1]: Can we use [jekyll-datapage_gen](https://github.com/avillafiorita/jekyll-datapage_gen) instead? If not, this plugin should be included in the [legumeinfo-theme repo](https://github.com/legumeinfo/legumeinfo-jekyll-theme), not the site repo.

File: _plugins/template_genus_index.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline _plugins/template_genus_index.html:1]: This file should reside in _layouts.

File: taxa/Aeschynomene/index.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Aeschynomene/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Apios/index.html [hunk 5]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Apios/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Arachis/index.html [hunk 6]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Arachis/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Bauhinia/index.html [hunk 7]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Bauhinia/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Cajanus/index.html [hunk 8]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Cajanus/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Cercis/index.html [hunk 9]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Cercis/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Chamaecrista/index.html [hunk 10]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Chamaecrista/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Cicer/index.html [hunk 11]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Cicer/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Faidherbia/index.html [hunk 12]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Faidherbia/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Glycine/index.html [hunk 13]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Glycine/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Lotus/index.html [hunk 14]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Lotus/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Lupinus/index.html [hunk 15]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Lupinus/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Medicago/index.html [hunk 16]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Medicago/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Phaseolus/index.html [hunk 17]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Phaseolus/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Pisum/index.html [hunk 18]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Pisum/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Trifolium/index.html [hunk 19]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Trifolium/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/Zenia/index.html [hunk 20]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (alancleary) [inline taxa/Zenia/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

File: taxa/index.html [hunk 21]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (alancleary) [inline taxa/index.html:1]: Pages generated by a plugin should not be included in the repository. The plugin should generate them at compile time and include them in the generated site located in the _site directory.

**Review by alancleary — CHANGES_REQUESTED** [review]
Overall I think these changes are great. I especially like how you organized the taxa. However, I think we need to tweak the plugin that generates the taxa and remove pages it generates from the repository since those should be generated with the rest of the site at compile time. See my comments for details.

File: _plugins/jekyll-generate-taxa.rb (hunk 2 above)
Comment (cann0010) [reply _plugins/jekyll-generate-taxa.rb:1]: Thanks for the review and the comments! I'll try to make the changes.
I am open to using jekyll-datapage_gen. I am not sure the best way to do that. Do you want to add that to the legumeinfo-theme repo?

**Comment by alancleary** [comment]
> Thanks for the review and the comments! I'll try to make the changes.
I am open to using jekyll-datapage_gen. I am not sure the best way to do that. Do you want to add that to the legumeinfo-theme repo?

@cann0010 Yes, I think adding it to the legumeinfo-theme repo would be best. Option 1 (as described in the [jekyll-datapage_gen readme](https://github.com/avillafiorita/jekyll-datapage_gen/blob/master/README.org)) is probably the way to go, i.e. add the gem to the theme's gemfile and then list it as a plugin and configure it in the theme config file.

**Comment by cann0010** [comment]
I am closing this pull request. I have tried to make the changes instead on branch dlt-jeklis-datapage. I'll initiate a PR from that branch.

**Pull request closed without merging**